backend/src/test/java/de/kathahuth/gymast/service/GroupServiceTest.java

```diff
@@ -6,31 +6,82 @@
 import org.junit.jupiter.api.Test;
 
 import java.util.List;
+import java.util.Optional;
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 class GroupServiceTest {
 
-    private final GroupMongoDb groupDb = mock(GroupMongoDb.class);
-    private final GroupService groupService = new GroupService(groupDb);
+    private final GroupMongoDb mockGroupMongoDb = mock(GroupMongoDb.class);
+    private final GroupService groupService = new GroupService(mockGroupMongoDb);
+
+    private Group createDynamicGroup() {
+        return Group.builder()
+                .id("Dynamic Strength")
+                .lable("difficulty")
+                .build();
+    }
+
+    private Group createJumpGroup() {
+        return Group.builder()
+                .id("Jump and Leaps")
+                .lable("difficulty")
+                .build();
+    }
 
     @Test
     @DisplayName("List group should return list of difficulty group from mongo db")
     public void listGroups(){
+
         //Given
-        when(groupDb.findAll()).thenReturn(List.of(
-                new Group("difficultygroup1"),
-                new Group("difficultygroup2")));
+        when(mockGroupMongoDb.findAll()).thenReturn(List.of(
+                createDynamicGroup(),
+                createJumpGroup()));
 
         //When
         List<Group> groups = groupService.listGroups();
 
         //Then
         assertThat(groups, containsInAnyOrder(
-                new Group("difficultygroup1"),
-                new Group("difficultygroup2")));
+                createDynamicGroup(),
+                createJumpGroup()));
+    }
+
+    @Test
+    @DisplayName("getGroupById should return group by Id")
+    public void getGroupById(){
+
+        //Given
+        String groupId = "Dynamic Strength";
```

Comment:
if you are using descriptive Strings for your ID I would rather switch to an ID attribute without any space (IDs usually do not include spaces) e.g. `dynamicStrength` and include a description attribute you can display in your frontend